src/datachain/data_storage/sqlite.py

```diff
@@ -391,6 +396,9 @@ def __init__(
         # foreign keys
         self.default_table_names: list[str] = []
 
+        if in_memory:
```

Comment:
[Q] is it possible that a `db_file` has been provided and has been overwritten? Should an error be thrown under those circumstances?

Reply:
Yep, added a function to check for this error case, and reduce code duplication, and added a test for this as well.